src/include/daos/tse.h

```diff
@@ -14,15 +14,13 @@
 #define __TSE_SCHEDULE_H__
 
 #include <gurt/list.h>
-/**
- * tse_task is used to track single asynchronous operation.
- * 1K bytes all together.
- */
-#define TSE_TASK_SIZE		1024
-/* 8 bytes used for public members */
-#define TSE_PRIV_SIZE		1016
+
 /* tse_task arguments max length */
-#define TSE_TASK_ARG_LEN		880
+#define TSE_TASK_ARG_LEN	(840 + sizeof(pthread_mutex_t))
```

Comment:
I just feel this looks wired, why need to tight task arg len with size of mutex, at lest need details comments, and looks not very good to add such thing here for a very common used task length definition. (the size problem on arm is hit on at cli_obj.c)

Another option is at cli_obj.c reduce "OBJ_TGT_INLINE_NR" for ARM platform with comment explain the size difference of mutex used in "struct obj_reasb_req" on ARM platform. 
just FYI.

Reply:
Well the issue is that originally this didn’t look good as it was hardcoded. So this is just making it more dynamic as the pthread mutex is of different size. I can add a comment that pthread mutex is of different size on different platforms to make this clearer. I don’t have other suggestions. Changing OBJ_TGT_INLINE_NR is just doing the same thing really in a different place.

Reply:
if you prefer i can revert to the original change adding the:
#ifdef __aarch64__
and just hardcode things.
but i was just addressing the suggestion from @jolivier23 to just use sizeof()

Reply:
hi, i just feel looks not very easy to understand what's the reason of "#define TSE_TASK_ARG_LEN	(840 + sizeof(pthread_mutex_t))" in tse.h, as some tasks even don't use any mutex.
As the real problem is introduced by cli_obj.c's compiling, probably fine to fix it there.
But not a big deal, can leave as it and later after we cleanup things in cli obj we may change it back. thanks.

Reply:
the suggestion is to only change the task in cli_obj.c which actually caused the compiling failure, to avoid confuse to change a very commonly used task struct's length.

Reply:
im a little against the change in cli_obj.c because it would change the behavior on the client between 2 different architectures. we can change it for both, but in this case for x86, then the task buf size is not in excess. and if someone adds more fields later and forgets to bump the task buf size, it would still work on x86 but would break aarch64. so i think this needs to be account for in the task buf size.